test/table.js

```diff
@@ -1312,6 +1312,19 @@ describe('Bigtable/Table', function() {
     });
   });
 
+  describe('waitForReplication', function() {
```

Comment:
Here are some tests we should have for this PR:

- it should generate a token (check `client`, `method`, and `reqOpts` args for accuracy)
- it should return error if the token could not be generated (line 1379)

I recommend moving `checkConsistency` to its own method, i.e. `Table.prototype.checkConsistency`. After that, you can test:

- it should check consistency (check `client`, `method`, and `reqOpts` args for accuracy)
- it should return error if consistency check failed
- it should return true if consistent
- it should return false if not consistent

Back to the `waitForReplication` tests:

- it should set a timeout to return false after 10 minutes
- it should check if consistent (did it call the new method, `checkConsistency` correctly)
- it should return true if consistent
- it should clear timeout if consistent
- it should try again after 5 seconds if not consistent

I believe that covers it all!

Reply:
@vijay-qlogic: what's the status of these tests?